src/AddIns/Uno.UI.MediaPlayer.WebAssembly/HtmlMediaPlayer.cs

```diff
@@ -473,22 +482,45 @@ internal void UpdateVideoStretch(Stretch stretch)
 
 	public void Play()
 	{
+
+		TimeUpdated += OnHtmlTimeUpdated;
 		if (this.Log().IsEnabled(LogLevel.Debug))
 		{
 			this.Log().Debug($"Play()");
 		}
+		if (ActiveElement != null && !_isPlaying)
+		{
+			NativeMethods.Play(ActiveElement.HtmlId);
+			_isPlaying = true;
+		}
+	}
 
-		NativeMethods.Play(ActiveElement.HtmlId);
+	public void Pause()
+	{
+		TimeUpdated -= OnHtmlTimeUpdated;
+		if (this.Log().IsEnabled(LogLevel.Debug))
+		{
+			this.Log().Debug($"Pause()");
+		}
+		if (ActiveElement != null && _isPlaying)
+		{
+			NativeMethods.Pause(ActiveElement.HtmlId);
+			_isPlaying = false;
+		}
 	}
 
 	public void Stop()
 	{
+		TimeUpdated -= OnHtmlTimeUpdated;
```

Comment:
Is this method called when the playback automatically stops at the end? (So that TimeUpdated is unregistered as well)

Reply:
I had to remove the OnHtmlTimeUpdated.
Because when we don't have a source or when it's changing, the player (html video) doesn't have the currentTime and it was giving an error in the javascript. What prevented changing the sources and making the controls work after the first error.
So I am adding  it in play, and remove it in pause and stop.
And in the case of Wasm, when the loop is activated, it doesn't stop.
What you said happened at the end without a loop. I had to update the _isPlaying=false when the video endded. 
Good catch. Thanks for review.